app/javascript/packs/mock-device-profiling.tsx

```diff
@@ -0,0 +1,124 @@
+import { render } from 'react-dom';
+import { useInstanceId } from '@18f/identity-react-hooks';
+import { ChangeEvent } from 'react';
+
+/**
+ * Loads the ?session_id=SESSION_ID param from the <script> tag. Unfortunately
+ * import.meta.url doesn't have live URL params so we need to scrape the DOM.
+ */
+function loadSessionId(): string | undefined {
+  let sessionId;
+  Array.from(document.scripts).every((scriptTag) => {
+    if (scriptTag.src.includes('session_id')) {
+      sessionId = new URL(scriptTag.src).searchParams.get('session_id');
+      return false;
+    }
+    return true;
+  });
+  return sessionId;
+}
+
+function submitMockFraudResult({ result, sessionId }: { result: string; sessionId?: string }) {
+  window.fetch('/test/device_profiling', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      session_id: sessionId,
+      result,
+    }),
+  });
+}
+
+const CHAOS_OPTIONS: { apply: () => void; undo: () => void }[] = [
+  {
+    apply: () => {
+      document.body.style.transform = 'scale(-1, 1)';
+    },
+    undo: () => {
+      document.body.style.transform = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.transform = 'rotate(5deg)';
+    },
+    undo: () => {
+      document.body.style.transform = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.filter = 'invert(100%)';
+    },
+    undo: () => {
+      document.body.style.filter = '';
+    },
+  },
+  {
+    apply: () => {
+      document.body.style.fontFamily = 'Comic Sans MS';
+    },
+    undo: () => {
+      document.body.style.fontFamily = '';
+    },
+  },
+];
+
+let undoLast: () => void | undefined;
```

Comment:
Since we're implementing this with React, this seems like it'd be a good use-case for `useState` and `useEffect`.

Something like...

```tsx
const [value, setValue] = useState('');
const [undoLast, setUndoLast] = useState<undefined | () => void>();
useEffect(() => undoLast?.(), [value]); 
```

Reply:
oh duh, great call! and with `useEffect` we can even return the undo function at the end of the hook to help clean up as well

Reply:
updated 67f2584fd65805afa5dc8de427f23e046317d7ba